repo_commit.go

```diff
@@ -108,6 +108,31 @@ func (repo *Repository) getCommitIdOfPackedRef(refpath string) ([]byte, error) {
 	return nil, RefNotFound(refpath)
 }
 
+type packedRef struct {
+	commit  string
+	refpath string
+}
+
+// parseRefLine parses a line in the packed-refs file. This file
+// contains lines of the form `${commit-id} ${ref-name}`,
+// `^${commit-id}, and comment lines beginning with "#". This function
+// returns the parsed ref in the first case and an error in all other
+// cases.
+func parseRefLine(line string) (packedRef, error) {
+	parseErr := fmt.Errorf("could not parse ref from line %q", line)
+	fields := strings.Fields(line)
+	if len(fields) != 2 {
+		return packedRef{}, parseErr
+	}
+	if len(fields[0]) != 40 {
```

Comment:
> It ignores comment lines beginning with "#"

Can you point which line(s) is responsible for that, I don't see where that happens.